test/integration_test.go

```diff
@@ -0,0 +1,96 @@
+package test //nolint:testpackage // we want to use currently private functions in the test package.
+
+import (
+	"bytes"
+	_ "embed"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/snyk/go-application-framework/pkg/configuration"
+)
+
+var (
+	//go:embed testdata/opensource_scan_output.txt
+	openSourceScanOutput []byte
+	//go:embed testdata/opensource_scan_depgraph.json
+	openSourceDepGraph []byte
+
+	//go:embed testdata/container_scan_output.txt
+	containerScanOutput []byte
+	//go:embed testdata/container_scan_depgraph.json
+	containerDepGraph []byte
+)
+
+func TestIntegration(t *testing.T) {
+	type testCase struct {
+		command           string
+		scanResult        []byte
+		expectedDepGraphs []byte
+		config            map[string]any
+	}
+	tcs := []testCase{{
+		command:           "sbom",
+		scanResult:        openSourceScanOutput,
+		expectedDepGraphs: openSourceDepGraph,
+		config: map[string]any{
+			// doesn't matter much.
+			"targetDirectory": "./testdata/container_scan_output.txt",
+		},
```

Comment:
Obviously we could also test a combination of different flags & whatnot, but as the underlying scan is mocked away, I'm not sure if that really makes sense :)